docs/source/tertiary/whole-genome-regression.rst

```diff
@@ -0,0 +1,352 @@
+===============================
+GlowGR: Whole-Genome Regression
+===============================
+
+.. invisible-code-block: python
+
+    import glow
+    glow.register(spark)
+
+    genotypes_vcf = 'test-data/gwas/genotypes.vcf.gz'
+    covariates_csv = 'test-data/gwas/covariates.csv.gz'
+    continuous_phenotypes_csv = 'test-data/gwas/continuous-phenotypes.csv.gz'
+
+Glow supports Whole Genome Regression (WGR) as GlowGR, a parallelized version of the
+`regenie <https://rgcgithub.github.io/regenie/>`_ method (see the
+`preprint <https://www.biorxiv.org/content/10.1101/2020.06.19.162354v1>`_).
+
+.. image:: ../_static/images/wgr_runtime.png
+   :scale: 50 %
+
+GlowGR consists of the following stages.
+
+- Blocking the genotype matrix across samples and variants.
+- Performing dimension reduction with ridge regression.
+- Estimating phenotypic values with ridge regression.
+
+----------------
+Data preparation
+----------------
+
+GlowGR uses three basic input datasources.
+
+Genotype data
+=============
+
+The genotype data may be read from any variant datasource supported by Glow, such as VCF, BGEN or PLINK. The DataFrame
+must also include a column ``values`` containing a numeric representation of each genotype. The genotypic values may
+not be missing, or equal for every sample in a variant.
+
+Example
+-------
+
+When loading in the variants, perform the following transformations:
+
+- Split multiallelic variants with the ``split_multiallelics`` transformer.
+- Calculate the number of alternate alleles for biallelic variants with ``glow.genotype_states``.
+- Replace any missing values with the mean of the non-missing values using ``glow.mean_substitute``.
+- Filter out all homozygous SNPs.
+
+.. code-block:: python
+
+    from pyspark.sql.functions import col, lit
+
+    variants = spark.read.format('vcf').load(genotypes_vcf)
+    genotypes = glow.transform('split_multiallelics', variants) \
+        .withColumn('values', glow.mean_substitute(glow.genotype_states(col('genotypes')))) \
+        .filter('size(array_distinct(values)) > 1') \
+
+Phenotype data
+==============
+
+The phenotype data is represented as a Pandas DataFrame indexed by the sample ID. Each column represents a single
+phenotype. It is assumed that there are no missing phenotype values, and that the phenotypes are mean centered at 0.
```

Comment:
oh, how are missing phenos treated? Does the algorithm fail? What if you are running 50 phenos, and each has some missing phenos? Do you have to filter all samples with one or more missing phenos?

Reply:
Historically, I believe we've filtered out samples with missing phenotypic values in the single-phenotype setting.